ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/rest/WebServiceRequest.java

```diff
@@ -96,150 +87,156 @@ public Object request(String url, RequestMethod method, Map<String, Object> data
     }
 
     private Object get(String url) throws ClientException, IOException {
-        HttpGet request = new HttpGet(url);
-        request.setHeader(HEADER_KEY, mSubscriptionKey);
 
-        HttpResponse response = this.mClient.execute(request);
-        int statusCode = response.getStatusLine().getStatusCode();
-        if (statusCode == HttpStatus.SC_OK) {
-            return readInput(response.getEntity().getContent());
+        Request request = new Request.Builder()
+                .url(url)
+                .get()
+                .header(HEADER_KEY, mSubscriptionKey)
+                .build();
+
+        Response response = this.mClient.newCall(request).execute();
+        if (response.isSuccessful()) {
+            return readInput(response);
         } else {
-            String json = readInput(response.getEntity().getContent());
+            String json = readInput(response);
             if (json != null) {
                 ServiceError error = mGson.fromJson(json, ServiceError.class);
                 if (error != null) {
                     throw new ClientException(error.error);
                 }
             }
 
-            throw new ClientException("Error executing GET request!", statusCode);
+            throw new ClientException("Error executing GET request!", response.code());
         }
     }
 
 
     private Object patch(String url, Map<String, Object> data, String contentType) throws ClientException, IOException {
-        HttpPatch request = new HttpPatch(url);
-        request.setHeader(HEADER_KEY, mSubscriptionKey);
+
         String json = mGson.toJson(data);
-        StringEntity entity = new StringEntity(json, "utf-8");
-        request.setEntity(entity);
-        request.setHeader(CONTENT_TYPE, APPLICATION_JSON);
-        HttpResponse response = mClient.execute(request);
-
-        int statusCode = response.getStatusLine().getStatusCode();
-        if (statusCode == HttpStatus.SC_OK) {
-            return readInput(response.getEntity().getContent());
+
+        Request.Builder builder = new Request.Builder()
+                .url(url)
+                .header(HEADER_KEY, mSubscriptionKey)
```

Comment:
[Nit] Will `.addHeader(HEADER_KEY, this.mSubscriptionKey)` work? Let's keep it consistent with below `Request.Builder`s.

Reply:
[Nit] Also let's unify the code of `Request.Builder builder` and `Request request = //... .build()`

Reply:
addHeader allows having multiple headers with the same key, while header replace existing headers with the same key.
I think that in this situation "header" is better; I'll replace addHeader with header

Reply:
Agreed.